iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/newtype/arithmetic.hpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef IOX_HOOFS_CXX_NEWTYPE_ARITHMETIC_HPP
+#define IOX_HOOFS_CXX_NEWTYPE_ARITHMETIC_HPP
+#include "iceoryx_hoofs/internal/cxx/newtype/internal.hpp"
+
+namespace iox
+{
+namespace cxx
+{
+namespace newtype
+{
+template <typename T>
+// not required since a default'ed destructor does not define a destructor, hence the copy/move operations are
+// not deleted.
+// the only adaptation is that the dtor is protected to prohibit the user deleting the child type by
+// explicitly calling the destructor of the base type.
+// NOLINTNEXTLINE(cppcoreguidelines-special-member-functions, hicpp-special-member-functions)
+struct Arithmetic
+{
+  protected:
+    ~Arithmetic() = default;
+};
+
+template <typename T>
+auto operator+(const T& rhs, const T& lhs) noexcept -> typename T::value_type
```

Comment:
Just for completeness sake. I would add here
* `operator /=`
* `operator *=`
* `operator +=`
* `operator -=`

Since they are also arithmetic operators and as a naive user I would expect the newtype to have those operators when I add `arithemtic.`

Reply:
Guys, we just deleted/moved code to `dust` since we wanted to concentrate on the certification effort for the existing code which is used by `posh`. Now you want to add more code to `hoofs` which isn't used anywhere?

Reply:
btw, since we talked about a type builder, I had a look at the implementation of `add` and noticed that this PR breaks the `NewType` ... more after my vacation ;)

Reply:
@elBoberido Solved the above issue :heavy_check_mark: Can you have a look?